src/NextGen/Framework/Blocks/BlockModel.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Give\NextGen\Framework\Blocks;
+
+use Give\Framework\Support\Contracts\Arrayable;
+
+/**
+ * @unreleased
+ * A structured model for a Gutenberg block.
+ * Similar to WP_Block_Parser_Block, but without innerHTML or innerContent.
+ * ... and without the HTML comments as structure.
+ * ... but now with a Collection for innerBlocks :)
+ */
+class BlockModel implements Arrayable
+{
+    /** @var string */
+    public $name;
+
+    /** @var string */
+    public $clientId;
+
+    /** @var bool */
+    public $isValid;
+
+    /** @var array */
+    public $attributes;
+
+    /** @var BlockCollection */
+    public $innerBlocks;
```

Comment:
Whichever ones can be private, let's do so to close up scope.

Reply:
They all need to be public in order to serialize the block appropriately.

Reply:
Why doesn't the class itself handle its own serialization?

Reply:
Right now, we're only serializing the collection of block models and exposing each block model as an array of properties.  At this point everything can be public.